x-pack/plugins/ml/public/application/util/custom_url_utils.ts

```diff
@@ -76,15 +76,20 @@ export function getUrlForRecord(
 // Opens the specified URL in a new window. The behaviour (for example whether
 // it opens in a new tab or window) is determined from the original configuration
 // object which indicates whether it is opening a Kibana page running on the same server.
-// fullUrl is the complete URL, including the base path, with any dollar delimited tokens
-// from the urlConfig having been substituted with values from an anomaly record.
-export function openCustomUrlWindow(fullUrl: string, urlConfig: UrlConfig) {
+// `url` is the URL with any dollar delimited tokens from the urlConfig
+// having been substituted with values from an anomaly record.
+export function openCustomUrlWindow(url: string, urlConfig: UrlConfig, basePath: string) {
   // Run through a regex to test whether the url_value starts with a protocol scheme.
   if (/^(?:[a-z]+:)?\/\//i.test(urlConfig.url_value) === false) {
-    window.open(fullUrl, '_blank');
+    // If `url` is a relative path, we need to prefix the base path.
+    if (url.charAt(0) !== '/') {
+      url = `${basePath}${isKibanaUrl(urlConfig) ? '/app/' : '/'}${url}`;
```

Comment:
Looks like we need to add the security app URLs to this check. Currently this is trying to open up e.g.

```
1:46
http://localhost:5601/pgy/security/hosts/ml-hosts?_g=()&query=(query:%27process.name%20:%20%22filebeat%22%27,language:kuery)&timerange=(global:(linkTo:!(timeline),timerange:(from:%272019-08-20T23:00:00.000Z%27,kind:absolute,to:%272019-08-21T22:59:59.999Z%27)),timeline:(linkTo:!(global),timerange:(from:%272019-08-20T23:00:00.000Z%27,kind:absolute,to:%272019-08-21T22:59:59.999Z%27)))
```
which is missing the `app` part before `security`.

![image](https://user-images.githubusercontent.com/7405507/88914781-0a9ade80-d25b-11ea-8847-1b9a514d4b12.png)

Reply:
Pushed a fix in 1e64ab3.